container_opts.go

```diff
@@ -127,3 +127,12 @@ func setSnapshotterIfEmpty(c *containers.Container) {
 		c.Snapshotter = DefaultSnapshotter
 	}
 }
+
+// WithContainerExtension appends extension data to the container object.
+// Use this to decorate the container object with additional data.
+func WithContainerExtension(extension types.Any) NewContainerOpts {
```

Comment:
Could this take an interface and we do the any marshal in the opt?

Reply:
Done